vault/identity_store_util.go

```diff
@@ -479,9 +493,11 @@ LOOP:
 				// Ensure that there are no entities with duplicate names
 				entityByName, err := i.MemDBEntityByName(nsCtx, entity.Name, false)
 				if err != nil {
+					tx.Abort()
```

Comment:
It looks like [txn.Abort()](https://pkg.go.dev/github.com/hashicorp/go-memdb#Txn.Abort) handles already-committed or Aborted transactions, so it's probably safer to just defer this call immediately after starting the transaction. Fewer SLOC and protects us from future bugs.

Reply:
I agree with Mike here and had the same thought while reviewing the PR.

Reply:
Ugh I guess it's not possible to defer, Mike :( Maybe that's why Mia decided against it. Because the whole logic is surrounded by a for-loop.

Reply:
Yup, that'd do  it. Sorry about that!

Reply:
We could probably wrap this in a function move the logic (with defer) there, but maybe not worth it...

Reply:
yup, i agree with this whole thread. I wrapped the call in a function and switched to a defer.